core/chains/evm/log/broadcaster.go

```diff
@@ -519,17 +529,22 @@ func (b *broadcaster) invalidatePool() int64 {
 	return -1
 }
 
-func (b *broadcaster) onNewLog(log types.Log) {
+func (b *broadcaster) onNewLog(log types.Log) (shouldResubscribe bool) {
 	b.maybeWarnOnLargeBlockNumberDifference(int64(log.BlockNumber))
 
 	if log.Removed {
 		// Remove the whole block that contained this log.
 		b.logger.Debugw("Found reverted log", "log", log)
 		b.logPool.removeBlock(log.BlockHash, log.BlockNumber)
-		return
+		// this can happen with buggy RPC's, the only fix is to replay from the block before the reverted log
+		if (log.BlockHash == common.Hash{}) || log.BlockNumber == 0 {
```

Comment:
Ideally, we should add a test for this case

Reply:
Agreed, looking at how to add a test.